cmake/GoogleCloudCppLibrary.cmake

```diff
@@ -0,0 +1,197 @@
+# ~~~
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ~~~
+
+# A function to add targets for GA libraries that use gRPC for transport.
+#
+# * library:      the short name of the library, e.g. `kms`.
+# * display_name: the display name of the library, e.g. "Cloud Key Management
+#   Service (KMS) API"
+# * GOOGLE_CLOUD_CPP_SERVICE_DIRS: a list of service directories within the
+#   library.
+#
+function (google_cloud_cpp_add_ga_grpc_library library display_name)
+    set(library_target "google_cloud_cpp_${library}")
+    set(mocks_target "google_cloud_cpp_${library}_mocks")
+    set(protos_target "google_cloud_cpp_${library}_protos")
```

Comment:
```suggestion
function (google_cloud_cpp_add_ga_grpc_library library display_name)
    cmake_parse_arguments(_opt "" "" "SERVICE_DIRS" ${ARGN})
    set(library_target "google_cloud_cpp_${library}")
    set(mocks_target "google_cloud_cpp_${library}_mocks")
    set(protos_target "google_cloud_cpp_${library}_protos")
```

Reply:
Hm, so I tried this but `cmake_parse_arguments(...)` ignores the empty string as an argument. So it would not work for us.

```diff
diff --git a/cmake/GoogleCloudCppLibrary.cmake b/cmake/GoogleCloudCppLibrary.cmake
 function (google_cloud_cpp_add_ga_grpc_library library display_name)
+    cmake_parse_arguments(_opt "" "" "SERVICE_DIRS" ${ARGN})
     set(library_target "google_cloud_cpp_${library}")
     set(mocks_target "google_cloud_cpp_${library}_mocks")
     set(protos_target "google_cloud_cpp_${library}_protos")
@@ -36,7 +36,8 @@ function (google_cloud_cpp_add_ga_grpc_library library display_name)

     unset(mocks_globs)
     unset(source_globs)
-    foreach (dir IN LISTS GOOGLE_CLOUD_CPP_SERVICE_DIRS)
+    foreach (dir IN LISTS _opt_SERVICE_DIRS)
+        message(STATUS "DARREN: dir=${dir}")
         string(REPLACE "/" "_" ns "${dir}")
         list(APPEND source_globs "${dir}*.h" "${dir}*.cc" "${dir}internal/*")
         list(APPEND mocks_globs "${dir}mocks/*.h")
@@ -46,6 +47,7 @@ function (google_cloud_cpp_add_ga_grpc_library library display_name)
                  "${CMAKE_CURRENT_SOURCE_DIR}/${dir}samples")
         endif ()
     endforeach ()
+    message(FATAL_ERROR "breaking...")

diff --git a/google/cloud/kms/CMakeLists.txt b/google/cloud/kms/CMakeLists.txt
@@ -16,10 +16,9 @@

 include(GoogleCloudCppLibrary)

-set(GOOGLE_CLOUD_CPP_SERVICE_DIRS "" "inventory/v1/" "v1/")
-
-google_cloud_cpp_add_ga_grpc_library(kms
-                                     "Cloud Key Management Service (KMS) API")
+google_cloud_cpp_add_ga_grpc_library(
+    kms "Cloud Key Management Service (KMS) API" SERVICE_DIRS ""
+    "inventory/v1/" "v1/")

 if (BUILD_TESTING AND GOOGLE_CLOUD_CPP_ENABLE_CXX_EXCEPTIONS)
     add_executable(kms_quickstart "quickstart/quickstart.cc")
```

Output:
```
-- Found RE2 via CMake.
-- CURL linkage will be non-static
-- CURL linkage will be non-static
-- DARREN: dir=inventory/v1/
-- DARREN: dir=v1/
CMake Error at cmake/GoogleCloudCppLibrary.cmake:50 (message):
  breaking...
```